x-pack/plugin/repository-encrypted/src/main/java/org/elasticsearch/repositories/encrypted/EncryptedRepository.java

```diff
@@ -233,16 +211,82 @@ public void finalizeSnapshot(
     }
 
     @Override
-    public void snapshotShard(SnapshotShardContext context) {
+    public String startVerification() {
+        // verification bypasses the encrypted blobstore because it works easier without a password generation
+        String seed = super.startVerification();
+        assert seed.indexOf('.') == -1;
+        String uuid = UUIDs.randomBase64UUID();
+        assert uuid.indexOf('.') == -1;
         try {
-            validateLocalRepositorySecret(context.userMetadata());
-        } catch (RepositoryException passwordValidationException) {
-            context.onFailure(passwordValidationException);
+            String computedHash = AESKeyUtils.computeId(AESKeyUtils.generatePasswordBasedKey(repositoryPassword, uuid));
```

Comment:
This is another instance where pbkdf2 is done on the calling thread because the caller is not async. In this case, as well as in the other repository verification methods, the generic thread pool ends up doing the crunch.
We haven't discussed this case explicitly, so feel free to discuss.
The alternative is to fork but wait for the result, as long as the `startVerification` method is not async (takes an action listener as an argument).